File: Python/Graph/Graph-Representation-Adjacency-List-2.py
 # 
 # @author 
 # aakash.verma
 # 
 # Output:
 # Adjacency List of Undirected Graph
 # |Delhi| => Mumbai Bangalore 
 # |Mumbai| => Delhi Hyderabad Bangalore 
 # |Bangalore| => Delhi Mumbai 
 # |Hyderabad| => Mumbai
 # 
 #

class Graph:

	def __init__(self, cities):
		self.cities = cities
		self.adjacency_list = dict()
		for city in self.cities:
			self.adjacency_list[city] = list()

	def add_edge(self, source, destination):
		self.adjacency_list[source].append(destination)
		self.adjacency_list[destination].append(source) # for directed graph comment this line

	def print_graph(self):
		print("Adjacency List of Undirected Graph")
		for city in self.cities:
			print("|", city, "| => ", end = " ")
			for nbr_city in self.adjacency_list[city]:
				print(nbr_city, end = " ")
			print()


if __name__ == '__main__':
	
	cities = ["Delhi", "Mumbai", "Bangalore", "Hyderabad"]
	g = Graph(cities);
	g.add_edge("Delhi", "Mumbai")
	g.add_edge("Delhi", "Bangalore")
	g.add_edge("Mumbai", "Hyderabad")
	g.add_edge("Mumbai", "Bangalore")
	g.print_graph()

File: Java/Graph/Graph-Representation-Adjacency-List-2.java
/** 
 * 
 * @author 
 * aakash.verma
 * 
 * Output:
 * Adjacency List of Undirected Graph
 * |Delhi| => Mumbai Bangalore 
 * |Mumbai| => Delhi Hyderabad Bangalore 
 * |Bangalore| => Delhi Mumbai 
 * |Hyderabad| => Mumbai
 * 
 */

import java.util.*;

class Graph {

    List<String> cities;
    HashMap<String, ArrayList<String>> adjacencyList; 
    
    public Graph(List<String> cities) {
        this.cities = cities;
        adjacencyList = new HashMap<>();

        for (String city : this.cities) {
            adjacencyList.put(city, new ArrayList<>());
        }
    }

    public void addEdge(String source, String destination) {
        this.adjacencyList.get(source).add(destination);
        this.adjacencyList.get(destination).add(source); // for directed graph comment this line
    }

    public void printGraph() {
        System.out.println("Adjacency List of Undirected Graph");
        for (String city : this.cities) {
            System.out.print("|" + city + "| => ");
            for(String nbrCity : adjacencyList.get(city)) {
                System.out.print(nbrCity + " ");
            }
            System.out.println();
        }
    }

    public static void main( String args[] ) {
        List<String> cities = new ArrayList<>();
        cities.add("Delhi");
        cities.add("Mumbai");
        cities.add("Bangalore");
        cities.add("Hyderabad");
        Graph g = new Graph(cities);
        g.addEdge("Delhi", "Mumbai");
        g.addEdge("Delhi", "Bangalore");
        g.addEdge("Mumbai", "Hyderabad");
        g.addEdge("Mumbai", "Bangalore");

        g.printGraph();
    }

}